src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private ColorSupplier color = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        String randomColor = color.getRandomColor();
+        final String[] figures = {"square",
+                                  "rectangle",
+                                  "right triangle",
+                                  "circle",
+                                  "isosceles trapezoid"};
+        String randomFigureType = figures[random.nextInt(figures.length)];
+
+        switch (randomFigureType) {
+            case "square":
+                double side = getRandomSquare();
+                return new Square(randomColor, side);
+            case "rectangle":
+                double length = getRandomSquare();
+                double width = getRandomSquare();
+                return new Rectangle(randomColor, length, width);
+            case "right triangle":
+                double firstLeg = getRandomSquare();
+                double secondLeg = getRandomSquare();
+                double hypotenuse = getRandomSquare();
+                return new RightTriangle(randomColor, firstLeg, secondLeg, hypotenuse);
+            case "circle":
+                double radius = getRandomSquare();
+                return new Circle(randomColor, radius);
+            case "isosceles trapezoid":
```

Comment:
was proposed to create a private method to encapsulate figure creational logic, e.g.

```
case "square":
  return getRandomSquare();
case "circle": 
  return getRandomCircle();
....
```

Reply:
Where should I create these methods??